functions/src/index.ts

```diff
@@ -34,12 +30,14 @@ exports.thingspeakToFirestore = functions
   .runWith(thingspeakToFirestoreRuntimeOpts)
   .pubsub.schedule('every 2 minutes')
   .onRun(async () => {
-    const sensorList = (await firestore.collection('/sensors').get()).docs;
+    const sensorDocQuerySnapshot = await firestore
+      .collection('sensors')
+      .where('isActive', '==', true)
+      .get();
 
-    for (const sensorDoc of sensorList) {
+    for (const sensorDoc of sensorDocQuerySnapshot.docs) {
       const sensorDocData = sensorDoc.data();
-      const isActive = sensorDocData.isActive ?? true;
-      if (isActive && sensorDocData.purpleAirId) {
+      if (sensorDocData.purpleAirId) {
```

Comment:
Why would a purple air ID be nullish?

Reply:
It should be, but this is just an extra check if there's a doc in the `sensors` collection that doesn't have a `purpleAirId` field